common/beerocks/scripts/prplmesh_utils.sh.in

```diff
@@ -315,7 +316,22 @@ main() {
             ;;
         "start_wait")
             start_function
-            wait -n
+
+            # Store PIDs of the Controller and Agents
+            PREV_PIDS=$(pgrep "beerocks_(agent|controller)")
+            CURR_PIDS=$PREV_PIDS
+            
+            # Monitor the number of running processes
+            while [ ${#CURR_PIDS[@]} -ge ${#PREV_PIDS[@]} ]; do
+                sleep 1
+                CURR_PIDS=$(pgrep "beerocks_(agent|controller)")
+                if [ ${#CURR_PIDS[@]} -gt ${#PREV_PIDS[@]} ]; then
+                        PREV_PIDS=$CURR_PIDS
+                fi
+            done
```

Comment:
This entire thing should be moved in a function. The main switch should just call functions that have the real logic.

Reply:
Moved